src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -823,26 +823,27 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 				// To get good reporting of errors we need to track the origin of the value for all method calls
 				// but except Newobj as those are special.
 				if (GetReturnTypeWithoutModifiers (calledMethodDefinition.ReturnType).MetadataType != MetadataType.Void) {
-					localReturnValue = GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
+					methodReturnValue = GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
 
 					return true;
 				}
 
 				return false;
 			}
 
-			//localReturnValue ??= GetReturnTypeWithoutModifiers (calledMethod.ReturnType).MetadataType == MetadataType.Void ? 
-				//MultiValueLattice.Top : GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
-
 			// If we get here, we handled this as an intrinsic.  As a convenience, if the code above
 			// didn't set the return value (and the method has a return value), we will set it to be an
 			// unknown value with the return type of the method.
-			if (localReturnValue.IsEmpty ()) {
-				if (GetReturnTypeWithoutModifiers (calledMethod.ReturnType).MetadataType != MetadataType.Void) {
-					localReturnValue = GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
-				}
+			maybeMethodReturnValue ??= GetReturnTypeWithoutModifiers (calledMethod.ReturnType).MetadataType == MetadataType.Void ? 
+				MultiValueLattice.Top : GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
+
+
+			if (maybeMethodReturnValue is not null) {
+				methodReturnValue = (MultiValue) maybeMethodReturnValue;
+			} else {
+				methodReturnValue = GetReturnTypeWithoutModifiers (calledMethod.ReturnType).MetadataType == MetadataType.Void ? 
+					MultiValueLattice.Top : GetMethodReturnValue (calledMethodDefinition, returnValueDynamicallyAccessedMemberTypes);
```

Comment:
This is weird - the above will make sure that `maybeMethodReturnValue` is never null (and it does it the exact same way)...

Reply:
Thanks for pointing this out, this needs to be changed